great_expectations/dataset/sqlalchemy_dataset.py

```diff
@@ -1191,11 +1187,15 @@ def create_temporary_table(self, table_name, custom_sql, schema_name=None):
                 table_name=table_name, custom_sql=custom_sql
             )
         elif self.sql_engine_dialect.name.lower() == "snowflake":
-            logger.info("Creating transient table %s" % table_name)
+            if not self.generated_table_name:
+                table_type = "TRANSIENT"
+            else:
+                table_type = "TEMPORARY"
+            logger.info("Creating temporary table %s" % table_name)
```

Comment:
Just a suggestion to convert to f-strings when changing/adding an interpolated string (https://www.python.org/dev/peps/pep-0498/)
```suggestion
            logger.info(f"Creating temporary table {table_name}")
```